Lib/idlelib/editor.py

```diff
@@ -1386,6 +1387,27 @@ def smart_indent_event(self, event):
         finally:
             text.undo_block_stop()
 
+    def smart_dedent_event(self, event):
+        # if selection:
+        #     do dedent-region
+        # elif only whitespace to the left:
+        #      dedent one level
+        first, last = self.get_selection_indices()
+        self.text.undo_block_start()
+        try:
+            if first and last:
+                if index2line(first) != index2line(last):
+                    return self.dedent_region_event(event)
+            prefix = self.text.get('insert linestart', 'insert')
+            raw, effective = classifyws(prefix, self.tabwidth)
```

Comment:
```suggestion
            raw, effective = get_line_indent(prefix, self.tabwidth)
```
This fixes the above issue.